scripts/dts/python-devicetree/src/devicetree/edtlib.py

```diff
@@ -225,14 +225,44 @@ def __init__(self, path: Optional[str], fname2path: Dict[str, str],
         self._check(require_compatible, require_description)
 
         # Initialize look up tables.
+        pattern_props = []
+        regular_props = []
+
         self.prop2specs: Dict[str, 'PropertySpec'] = {}
         for prop_name in self.raw.get("properties", {}).keys():
-            self.prop2specs[prop_name] = PropertySpec(prop_name, self)
+            prop_spec = PropertySpec(prop_name, self)
+            self.prop2specs[prop_name] = prop_spec
+            if prop_spec.pattern:
+                pattern_props.append(prop_spec)
+            else:
+                regular_props.append(prop_spec)
+
         self.specifier2cells: Dict[str, List[str]] = {}
         for key, val in self.raw.items():
             if key.endswith("-cells"):
                 self.specifier2cells[key[:-len("-cells")]] = val
 
+        # Merge pattern properties into matching regular properties.
+        for regular_prop in regular_props:
+            for pattern_prop in pattern_props:
+                if pattern_prop.pattern.match(regular_prop.name): # type: ignore
+                    # non-overridable properties
+                    regular_prop.type = pattern_prop.type
+                    regular_prop.deprecated = pattern_prop.deprecated
+                    regular_prop.specifier_space = pattern_prop.specifier_space
+                    # overridable properties
+                    if regular_prop.enum is None:
+                        regular_prop.enum = pattern_prop.enum
+                    if regular_prop.const is None:
```

Comment:
I think the comments should be swapped, and `pattern_prop` shouldn't override `regular_prop` with empty values.
```suggestion
                    # overridable properties
                    if pattern_prop.type:
                        regular_prop.type = pattern_prop.type
                    if pattern_prop.deprecated:
                        regular_prop.deprecated = pattern_prop.deprecated
                    if pattern_prop.specifier_space:
                        regular_prop.specifier_space = pattern_prop.specifier_space
                    # non-overridable properties
                    if regular_prop.enum is None:
                        regular_prop.enum = pattern_prop.enum
                    if regular_prop.const is None:
```

Reply:
btw, an overridable property is a property that, if defined in a matching property (e.g. `vin-supply` for `.*-supply$`) will override the values from the pattern property. non-overridable means that if set in the pattern property, it will be enforced.

Reply:
Okay, I thought it was the other way around, but I see that your view is reflected in the documentation.

For other dummies like me, maybe it would be more readable to reorder these items, and add more verbose comments:
```python
# Attributes taken from the pattern property
...

# Attributes constrained by the regular property
...
```